### ISSUE TITLE
ProfilePlot fails for particle fields

### ISSUE THREAD

**Issue by yt-fido** [yt-project/yt#733]
Originally reported by: **Nathan Goldbaum (Bitbucket: [ngoldbaum](https://bitbucket.org/ngoldbaum), GitHub: [ngoldbaum](https://github.com/ngoldbaum))**

----------------------------------------

```
#!python

In [4]: plot = ProfilePlot(pf.h.all_data(), 'ParticleRadiusMpc','particle_mass')
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
/Users/goldbaum/Desktop/yt-hg/yt/mods.pyc in <module>()
----> 1 plot = ProfilePlot(pf.h.all_data(), 'ParticleRadiusMpc','particle_mass')

/Users/goldbaum/Desktop/yt-hg/yt/visualization/profile_plotter.pyc in __init__(self, data_source, x_field, y_fields, weight_field, n_bins, accumulation, fractional, label, plot_spec, profiles)
    207                                             weight_field=weight_field,
    208                                             accumulation=accumulation,
--> 209                                             fractional=fractional)]
    210         else:
    211             self.profiles = ensure_list(profiles)

/Users/goldbaum/Desktop/yt-hg/yt/data_objects/profiles.pyc in create_profile(data_source, bin_fields, n, weight_field, fields, accumulation, fractional)
   1056     setattr(obj, "fractional", fractional)
   1057     if fields is not None:
-> 1058         obj.add_fields(fields)
   1059     for field in fields:
   1060         if fractional:

/Users/goldbaum/Desktop/yt-hg/yt/data_objects/profiles.pyc in add_fields(self, fields)
    817         temp_storage = ProfileFieldAccumulator(len(fields), self.size)
    818         for g in parallel_objects(self.data_source._grids):
--> 819             self._bin_grid(g, fields, temp_storage)
    820         self._finalize_storage(fields, temp_storage)
    821

/Users/goldbaum/Desktop/yt-hg/yt/data_objects/profiles.pyc in _bin_grid(self, grid, fields, storage)
    895
    896     def _bin_grid(self, grid, fields, storage):
--> 897         gd = self._get_data(grid, fields)
    898         if gd is None: return
    899         fdata, wdata, (bf_x,) = gd

/Users/goldbaum/Desktop/yt-hg/yt/data_objects/profiles.pyc in _get_data(self, grid, fields)
    854         # We want to make sure that our fields are within the bounds of the
    855         # binning
--> 856         filter, bin_fields = self._filter(bin_fields, pointI)
    857         if not np.any(filter): return None
    858         arr = np.zeros((bin_fields[0].size, len(fields)), dtype="float64")

/Users/goldbaum/Desktop/yt-hg/yt/data_objects/profiles.pyc in _filter(self, bin_fields, cut_points)
    838         # we also want to apply a filtering based on min/max
    839         filter = np.zeros(bin_fields[0].shape, dtype='bool')
--> 840         filter[cut_points] = True
    841         for (mi, ma), data in zip(self.bounds, bin_fields):
    842             filter &= (data > mi)

IndexError: too many indices for array
```

This was originally brought up on IRC by Sam.

Matt mentioned that this would not be worth the effort to solve in 2.x, although it should be straightforward to fix in 3.0 once `ProfilePlot` is merged into the 3.0 codebase.  For 2.x, I think we at least need a better error message, if that's possible.

----------------------------------------
- Bitbucket: https://bitbucket.org/yt_analysis/yt/issue/733
